flake.nix

```diff
@@ -0,0 +1,41 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.11";
+    flake-utils.url = "github:numtide/flake-utils";
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+      };
+    };
+  };
+  outputs =
+    {
+      self,
+      nixpkgs,
+      flake-utils,
+      rust-overlay,
+    }:
+    flake-utils.lib.eachDefaultSystem (
+      system:
+      let
+        overlays = [ (import rust-overlay) ];
+        pkgs = import nixpkgs {
+          inherit system overlays;
+        };
+      in
+      with pkgs;
+      {
+        devShells.default = mkShell {
+          buildInputs = [
+            picotool
+            (rust-bin.stable.latest.default.override {
```

Comment:
Probably better to pin a specific Rust version. We've already had CI suddenly start failing because Rust got a new release and it broke something

Reply:
It doesn't seem to pin Rust version directly, but I guess since it pins the overlay, what "`latest`" means will not change. I'm fine keeping it like this.